libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -63,68 +61,29 @@ void UARTDriver::begin(uint32_t b)
 
 void UARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
 {
-    if (device_path == NULL && _console) {
+    if (device_path == NULL && _console && !_initialised) {
```

Comment:
I think there is a better way to re-write this if/else since both are checking for `!_initialised`